#!/usr/bin/env python3
PACKAGE = "kinco_control"
from dynamic_reconfigure.parameter_generator import *

gen = ParameterGenerator()

gen.add("max_motor_rpm", int_t, 0, "Maximum motor RPM", 300, 0, 3000)
gen.add("gear_ratio", double_t, 0, "Gear ratio", 2.0, 1.0, 20.0)
gen.add("wheel_diameter", double_t, 0, "Wheel diameter (meters)", 0.2, 0.1, 1.0)
gen.add("wheels_x_distance", double_t, 0, "Distance between wheels in X (meters)", 0.0, 0.0, 1.0)
gen.add("wheels_y_distance", double_t, 0, "Distance between wheels in Y (meters)", 0.1, 0.0, 1.0)

# Real-time adjustable RPMs
gen.add("left_motor_rpm", int_t, 0, "RPM of the left motor", 0, -3000, 3000)
gen.add("right_motor_rpm", int_t, 0, "RPM of the right motor", 0, -3000, 3000)
gen.add("lift_motor_rpm", int_t, 0, "RPM of the right motor", 0, -1000, 1000)

exit(gen.generate(PACKAGE, "kinco_control", "Motor"))
